unlock-app/src/middlewares/web3Middleware.js

```diff
@@ -141,9 +141,16 @@ export default function web3Middleware({ getState, dispatch }) {
         // TODO: when the account has been updated we should reset web3Service and remove all listeners
         // So that pending API calls do not interract with our "new" state.
         web3Service.refreshAccountBalance(action.account)
-        web3Service.getPastLockCreationsTransactionsForUser(
-          action.account.address
-        )
+        web3Service
+          .getPastLockCreationsTransactionsForUser(action.account.address)
+          .then(lockCreations => {
+            // For each lock this user created, go get the history of interactions with that lock.
+            lockCreations.forEach(lockCreation => {
+              web3Service.getPastLockTransactions(
```

Comment:
We may want to only get the transactions on the lock for *that user* because I worry that for popular locks this maybe thousands of transactions... (but for now just add a TODO because we do not have that problem!)